lib/money/money.rb

```diff
@@ -361,10 +368,14 @@ def arithmetic(money_or_numeric)
     unless currency.compatible?(other.currency)
       Money.deprecate("mathematical operation not permitted for Money objects with different currencies #{other.currency} and #{currency}.")
     end
-    yield(other)
+    block_given? ? yield(other) : other
   end
 
   def calculated_currency(other)
     no_currency? ? other : currency
   end
+
+  def all_currencies(money_array)
+    (money_array + [self]).reject(&:no_currency?).map(&:currency).uniq
+  end
```

Comment:
This method extraction doesn't make much sense IMO.

Reply:
changed it around a little, let me know what you think, thx :)